cmd/agentctl2/utils/db_utils.go

```diff
@@ -0,0 +1,654 @@
+	Config *ProxyARPConfigWithMD
+}
+
+// IPScanNeighConfigWithMD contains a data record for interface configuration
+// and its etcd metadata.
+type IPScanNeighConfigWithMD struct {
+	Metadata  VppMetaData
+	IPScanNeighbor *l3.IPScanNeighbor
+}
+
+// IPScanNeighWithMD contains a data record for interface and its
+// etcd metadata.
+type IPScanNeighWithMD struct {
+	Config *IPScanNeighConfigWithMD
+}
+
+// NATConfigWithMD contains a data record for interface configuration
+// and its etcd metadata.
+type NATConfigWithMD struct {
+	Metadata  VppMetaData
+	Nat44Global *nat.Nat44Global
+}
+
+// NATWithMD contains a data record for interface and its
+// etcd metadata.
+type NATWithMD struct {
+	Config *NATConfigWithMD
+}
+
+// DNATConfigWithMD contains a data record for interface configuration
+// and its etcd metadata.
+type DNATConfigWithMD struct {
+	Metadata  VppMetaData
+	DNat44 *nat.DNat44
+}
+
+// DNATWithMD contains a data record for interface and its
+// etcd metadata.
+type DNATWithMD struct {
+	Config *DNATConfigWithMD
+}
+
+// IPSecPolicyConfigWithMD contains a data record for interface configuration
+// and its etcd metadata.
+type IPSecPolicyConfigWithMD struct {
+	Metadata  VppMetaData
+	SecurityPolicyDatabase *ipsec.SecurityPolicyDatabase
+}
+
+// IPSecPolicyWithMD contains a data record for interface and its
+// etcd metadata.
+type IPSecPolicyWithMD struct {
+	Config *IPSecPolicyConfigWithMD
+}
+
+// IPSecAssociationConfigWithMD contains a data record for interface configuration
+// and its etcd metadata.
+type IPSecAssosiciationConfigWithMD struct {
+	Metadata  VppMetaData
+	SecurityAssociation *ipsec.SecurityAssociation
+}
+
+// IPSecAssosiciationWithMD contains a data record for interface and its
+// etcd metadata.
+type IPSecAssosiciationWithMD struct {
+	Config *IPSecAssosiciationConfigWithMD
+}
+
+// lInterfaceConfigWithMD contains a data record for interface configuration
+// and its etcd metadata.
+type lInterfaceConfigWithMD struct {
+	Metadata  VppMetaData
+	Interface *linterface.Interface
+}
+
+// lInterfaceWithMD contains a data record for interface and its
+// etcd metadata.
+type lInterfaceWithMD struct {
+	Config *lInterfaceConfigWithMD
+}
+
+// lARPConfigWithMD contains a data record for interface configuration
+// and its etcd metadata.
+type lARPConfigWithMD struct {
+	Metadata  VppMetaData
+	ARPEntry *ll3.ARPEntry
+}
+
+// lARPWithMD contains a data record for interface and its
+// etcd metadata.
+type lARPWithMD struct {
+	Config *lARPConfigWithMD
+}
+
+// lRouteConfigWithMD contains a data record for interface configuration
+// and its etcd metadata.
+type lRouteConfigWithMD struct {
+	Metadata  VppMetaData
+	Route *ll3.Route
+}
+
+// lRouteWithMD contains a data record for interface and its
+// etcd metadata.
+type lRouteWithMD struct {
+	Config *lRouteConfigWithMD
+}
+
+// VppData defines a structure to hold all etcd data records (of all
+// types) for one VPP.
+type VppData struct {
```

Comment:
There actually is type `Config` which provides all of this configuration: https://github.com/ligato/vpp-agent/blob/731e9b4b9b2e37407e805221dcab9ccbd0482353/api/configurator/configurator.proto#L11-L14